addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1346,14 +1346,19 @@ def _notify_by_web_push_prepare_payload(self, message, msg_vals=False):
         payload = super()._notify_by_web_push_prepare_payload(message, msg_vals=msg_vals)
         payload['options']['data']['action'] = 'mail.action_discuss'
         record_name = msg_vals.get('record_name') if msg_vals and 'record_name' in msg_vals else message.record_name
+        author_id = [msg_vals.get('author_id')] if 'author_id' in msg_vals else message.author_id.ids
         if self.channel_type == 'chat':
-            author_id = [msg_vals.get('author_id')] if 'author_id' in msg_vals else message.author_id.ids
             payload['title'] = self.env['res.partner'].browse(author_id).name
             payload['options']['icon'] = '/discuss/channel/%d/partner/%d/avatar_128' % (message.res_id, author_id[0])
         elif self.channel_type == 'channel':
-            author_id = [msg_vals.get('author_id')] if 'author_id' in msg_vals else message.author_id.ids
             author_name = self.env['res.partner'].browse(author_id).name
             payload['title'] = "#%s - %s" % (record_name, author_name)
+        elif self.channel_type == 'group':
+            if not record_name:
+                member_names = self.channel_member_ids.mapped("partner_id.name")
+                record_name = f"{', '.join(member_names[:-1])} and {member_names[-1]}" if len(member_names) > 1 else member_names[0] if member_names else ""
+            author_name = self.env['res.partner'].browse(author_id).name
+            payload['title'] = "%s - %s" % (record_name, author_name)
```

Comment:
As a fix, looks decent :)

Would be nice to have a test coverage for it, as we'll likely clean the group chat name in master eventually (because this is computed both in JS and here in python, so it sucks to maintain 2 different parts of code that deal with same concern)